src/rest/handlers/rest_vertices.erl

```diff
@@ -105,7 +107,47 @@ to_json(Req, State = #{operation := listing}) ->
     {ok, Vertices} = graph:list_vertices(),
     {gmm_utils:encode(Vertices), Req, State}.
 
-
+%%% BULK handler
+
+-spec modify_state_bulk(Vertices :: list({binary(), binary()})) -> ok | {error, any()}.
+modify_state_bulk(Vertices)->
+    Parent = self(),
+
+    Pids = lists:map(fun({Type, Name}) ->
+        spawn(fun() ->
+            Result = try graph:create_vertex(Type, Name)
+                     catch Type:Reason:Stacktrace -> {'$pmap_error', self(), Type, Reason, Stacktrace} end,
+
+            Parent ! {self(), Result} end) end, Vertices),
+
+    % GATHERING RESULTS
+    Gather = fun
+    %PidsOrResults is initially the list of pids, gradually replaced by corresponding results
+                 F(PendingPids = [_ | _], PidsOrResults) ->
+                     receive
+                         {Pid, Result} ->
+                             NewPidsOrResults = rest_utils:replace(Pid, Result, PidsOrResults),
+                             F(lists:delete(Pid, PendingPids), NewPidsOrResults)
+                     after 5000 ->
+                         case lists:any(fun erlang:is_process_alive/1, PendingPids) of
+                             true ->
+                                 F(PendingPids, PidsOrResults);
+                             false ->
+                                 error({parallel_call_failed, {processes_dead, Pids}})
+                         end
+                     end;
+
+                 F([], AllResults) ->
+                     Errors = lists:filtermap(
+                         fun({'$pmap_error', Pid, Type, Reason, Stacktrace}) ->
+                             {true, {Pid, Type, Reason, Stacktrace}};
+                             (_) -> false end, AllResults),
+                     case Errors of
+                         [] -> ok;
+                         _ -> {error, Errors}
+                     end
+             end,
```

Comment:
Can do, but I would suggest moving `fun` keyword to next line, which would make overall depth of indentation smaller, again making code easier to read